pkg/csi/cinder/controllerserver.go

```diff
@@ -56,18 +56,30 @@ const (
 func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
 	klog.V(4).Infof("CreateVolume: called with args %+v", protosanitizer.StripSecrets(req))
 
-	// Volume cloud
-	volCloud := req.GetSecrets()["cloud"]
-	cloud, cloudExist := cs.Clouds[volCloud]
-	if !cloudExist {
-		return nil, status.Error(codes.InvalidArgument, "[CreateVolume] specified cloud undefined")
-	}
-
 	// Volume Name
 	volName := req.GetName()
 	volCapabilities := req.GetVolumeCapabilities()
 	volParams := req.GetParameters()
 
+	// Volume Type
+	volType := volParams["type"]
+
+	var volAvailability string
+	var volCloudAvailability string
+	if cs.Driver.withTopology {
+		// First check if volAvailability is already specified, if not get preferred from Topology
+		// Required, incase vol AZ is different from node AZ
+		volAvailability = volParams["availability"]
+		if volAvailability == "" {
+			accessibleTopologyReq := req.GetAccessibilityRequirements()
+			// Check from Topology
+			if accessibleTopologyReq != nil {
+				volCloudAvailability = sharedcsi.GetAZFromTopology(withTopologyKey, accessibleTopologyReq) // = region defined from the node label topology.kubernetes.io/region=region
+				volAvailability = sharedcsi.GetAZFromTopology(topologyKey, accessibleTopologyReq)          // = nova defined from Openstack
+			}
+		}
+	}
+
```

Comment:
Is there a reason we need to move this? It separates the assignment of `volName` and `volCapabilities` from the validation of same. Could you move it back?